qmat/solvers/generic/diffops.py

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Contains various specialized implementation of :class:`DiffOp` classes.
+"""
+import numpy as np
+from scipy.linalg import blas
+from typing import TypeVar
+
+from qmat.solvers.generic import DiffOp
+from qmat.utils import checkOverriding, storeClass
+
+T = TypeVar("T")
+
+
+DIFFOPS: dict[str, type[DiffOp]] = {}
+"""Dictionary containing all specialized :class:`DiffOp` classes"""
+
+def registerDiffOp(cls: type[T]) -> type[T]:
+    """Class decorator to register a specialized :class:`DiffOp` class in `qmat`"""
+    checkOverriding(cls, "evalF", isProperty=False)
+    storeClass(cls, DIFFOPS)
+    return cls
+
+
+@registerDiffOp
+class Dahlquist(DiffOp):
+    r"""
+    Implements a Dahlquist differential operator
+
+    .. math::
+
+        f(u,t) = \lambda u
+
+    Note
+    ----
+    This class is implemented for illustration and testing purposes.
+    For real applications, consider using the
+    :class:`qmat.solvers.dahlquist.Dahlquist` class instead.
```

Comment:
Lol, different people have different ideas of what a real application is, I guess :D